dot/peerset/peerset.go

```diff
@@ -729,3 +759,29 @@ func (ps *PeerSet) doWork(ctx context.Context) {
 		}
 	}
 }
+
+func (ps *PeerSet) periodicallyAllocateSlots(ctx context.Context) {
+	ticker := time.NewTicker(ps.nextPeriodicAllocSlots)
+
+	defer func() {
+		ticker.Stop()
+		close(ps.resultMsgCh)
+	}()
+
+	for {
+		select {
+		case <-ctx.Done():
+			err := ctx.Err()
+			if err != nil {
+				logger.Debugf("peerset slot allocation exiting: %s", err)
+			}
```

Comment:
this will always be not nil:

```suggestion
			logger.Debugf("peerset slot allocation exiting: %s", ctx.Err())
```